gr-blocks/lib/annotator_raw_impl.cc

```diff
@@ -42,20 +40,20 @@ void annotator_raw_impl::add_tag(uint64_t offset, pmt::pmt_t key, pmt::pmt_t val
     gr::thread::scoped_lock l(d_mutex);
 
     tag_t tag;
-    tag.srcid = pmt::intern(name());
+    tag.offset = offset;
     tag.key = key;
     tag.value = val;
-    tag.offset = offset;
+    tag.srcid = pmt::intern(name());
 
-    // add our new tag
-    d_queued_tags.push_back(tag);
-    // make sure our tags are in offset order
-    std::sort(d_queued_tags.begin(), d_queued_tags.end(), tag_t::offset_compare);
     // make sure we are not adding an item in the past!
-    if (tag.offset > nitems_read(0)) {
-        throw std::runtime_error(
-            "annotator_raw::add_tag: item added too far in the past.");
+    if (tag.offset < nitems_read(0)) {
+        throw std::runtime_error(fmt::format("annotator_raw::add_tag: item added too far "
```

Comment:
Should this even be an exception? Or should we maybe drop those items? (and maybe log-warn?)

Reply:
Hm, I think throwing here is fine, if you want to add a tag, you want to know whether that succeeded. Returning success or failure would make more sense, I agree, but if I change the behaviour of this block, this change can't be backported. We could make a separate PR to change the API to a more sensible `bool …::add_tag(uint64_t offset, pmt_t key, pmt_t val)` that doesn't throw, but it'd be a breaking change